phoenix-core/src/main/java/org/apache/phoenix/filter/PagedFilter.java

```diff
@@ -28,16 +28,21 @@
 import org.apache.hadoop.hbase.Cell;
 import org.apache.hadoop.hbase.CellUtil;
 import org.apache.hadoop.hbase.DoNotRetryIOException;
+import org.apache.hadoop.hbase.HConstants;
 import org.apache.hadoop.hbase.exceptions.DeserializationException;
 import org.apache.hadoop.hbase.filter.Filter;
 import org.apache.hadoop.hbase.filter.FilterBase;
 import org.apache.hadoop.hbase.util.Bytes;
 import org.apache.hadoop.hbase.util.Writables;
 import org.apache.hadoop.io.Writable;
 import org.apache.phoenix.util.EnvironmentEdgeManager;
+import org.apache.phoenix.util.ScanUtil;
 
 /**
- * This filter overrides the behavior of delegate so that we do not scan more rows than pageSizeInRows .
+ * This is a top level Phoenix filter which injected to a scan at the server side. If the scan has
+ * already a filter then PagedFilter wraps it. There two functions this filter implements: paging
+ * and TTL. The paging function makes sure that the scan does not take more than pageSizeInMs.
+ * The TTL function is for masking expired rows.
```

Comment:
Would it be a cleaner separation of concerns to stack two filters here, `PagedFilter` implements paging, `TTLFilter` implements masking of expired rows?

Reply:
I could not separate them. Both functionalities needs to be done at the top level filter. If we choose to skip an expired row under or above the PagedFilter then we cannot ensure paging happens timely. I am open to change the name of the class if you have a good suggestion for this.

Reply:
Isn't server-side paging a configurable feature?

Reply:
Yes, that is another good reason for separation.

Reply:
Instead of implementing masking in a filter, I implemented in a region scanner (PhoenixTLLRegionScanner) as the masking logic is more complex than what I thought initially.